core/lib/zksync_core/src/state_keeper/batch_executor/tests/mod.rs

```diff
@@ -69,11 +149,22 @@ impl SnapshotRecoveryMutation {
     }
 }
 
+const EXECUTE_L2_TX_AFTER_SNAPSHOT_RECOVERY_CASES: TestCases<(
+    Option<SnapshotRecoveryMutation>,
+    StorageType,
+)> = cases!(itertools::iproduct!(
+    SnapshotRecoveryMutation::ALL,
+    StorageType::ALL
+));
```

Comment:
Nit: There's [`Product`](https://docs.rs/test-casing/latest/test_casing/struct.Product.html) in `test-casing` that provides the same functionality without a need to use another crate.